validphys2/src/validphys/photon/compute.py

```diff
@@ -0,0 +1,265 @@
+            xgrid_reshape(eko, targetgrid = XGrid(self.xgrid), inputgrid = XGrid(self.xgrid))
+
+            # construct PDFs
+            pdfs = np.zeros((len(eko.rotations.inputpids), len(self.xgrid)))
+            for j, pid in enumerate(eko.rotations.inputpids):
+                if pid == 22 :
+                    pdfs[j] = photon_100GeV
+                    ph_id = j
+                if not self.qcd_pdfs[id].hasFlavor(pid):
+                    continue
+                pdfs[j] = np.array(
+                    [
+                        self.qcd_pdfs[id].xfxQ2(pid, x, self.q_in2) / x
+                        for x in self.xgrid
+                    ]
+                )
+            
+            # Apply EKO to PDFs
+            q2 = eko.mu2grid[0]
+            with eko.operator(q2) as elem:
+                pdf_final = np.einsum("ajbk,bk", elem.operator, pdfs)
+                # error_final = np.einsum("ajbk,bk", elem.error, pdfs)
+
+        photon_Q0 = pdf_final[ph_id]
+
+        # we want x * gamma(x)
+        return self.xgrid * photon_Q0
+    
+    def produce_interpolators(self):
+        self.photons_array = [self.compute_photon_array(i) for i in range(len(self.replicas_id))]
+        self.interpolator = [interp1d(self.xgrid, photon_array, fill_value=0., kind='cubic') for photon_array in self.photons_array]
+    
+    def compute(self, xgrid):
+        """
+        Compute the photon interpolating the values of self.photon_array.
+
+        Parameters
+        ----------
+        xgrid : nd.array
+            array of x values with shape (1,xgrid,1)
+        
+        Returns
+        -------
+        photon values : nd.array
+            array of photon values with shape (1,xgrid,1)
+        """
+        return [
+            self.interpolator[id](xgrid[0,:,0])[np.newaxis,:,np.newaxis]
+            for id in range(len(self.replicas_id))
+        ]
+    
+    def integrate(self):
+        """Compute the integral of the photon on the x range."""
+        return [trapezoid(self.photons_array[id], self.xgrid) for id in range(len(self.replicas_id))]
+    
+    def generate_error_matrix(self):
+        """generate error matrix to be used for the additional errors."""
+        if not self.fiatlux_runcard["additional_errors"] :
+            return None
+        extra_set = lhapdf.mkPDFs("LUXqed17_plus_PDF4LHC15_nnlo_100")
+        return np.array(
+            [
+                [(extra_set[i].xfxQ2(22, x, self.q_in2) - extra_set[0].xfxQ2(22, x, self.q_in2)) for i in range(101, 107+1)]
+                for x in self.xgrid
+            ]
+        ) # first index must be x, while second one must be replica index
+
+    def generate_errors(self):
+        """generate LUX additional errors."""
+        if self.error_matrix is None :
+            return np.zeros_like(self.xgrid)
+        u, s, _ = np.linalg.svd(self.error_matrix, full_matrices=False)
+        errors = u @ (s * np.random.normal(size=7))
```

Comment:
If this matters at all, we would want to have some control over the random state.

Reply:
Which kind of control?